packages/react/src/jsx/ReactJSXElementValidator.js

```diff
@@ -138,25 +143,30 @@ function validateExplicitKey(element, parentType) {
     }
     element._store.validated = true;
 
-    const currentComponentErrorInfo = getCurrentComponentErrorInfo(parentType);
+    const currentComponentErrorInfo = getCurrentComponentErrorInfo(
+      element._source,
```

Comment:
The idea was to remove the entire jsxDEV transform. So this PR is effectively also adding that whole transform back. That makes it a very large PR. Is it worth it?

Reply:
Remove when?\

Reply:
It’s only used for the “self” argument now. Since all string refs are warnings in 18+ that’s no longer needed in 18. So post-17 it’s no longer needed.